.github/workflows/deploy.yml

```
@@ -0,0 +1,40 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+    # Review gh actions docs if you want to further define triggers, paths, etc
+    # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on
+
```

Let's restrict the permissions to what's strictly needed so we don't have to worry as much about using external workflows.
```suggestion

permissions:
  contents: write

```